---
 libmultipath/dict.c |    9 +++++++++
 1 file changed, 9 insertions(+)

Index: multipath-tools-130222/libmultipath/dict.c
===================================================================
--- multipath-tools-130222.orig/libmultipath/dict.c
+++ multipath-tools-130222/libmultipath/dict.c
@@ -43,6 +43,9 @@ def_fast_io_fail_handler(vector strvec)
 	char * buff;
 
 	buff = set_value(strvec);
+	if (!buff)
+		return 1;
+
 	if (strlen(buff) == 3 && !strcmp(buff, "off"))
 		conf->fast_io_fail = MP_FAST_IO_FAIL_OFF;
 	else if (sscanf(buff, "%d", &conf->fast_io_fail) != 1 ||
@@ -1002,6 +1005,9 @@ hw_dev_loss_handler(vector strvec)
 	char * buff;
 	struct hwentry * hwe = VECTOR_LAST_SLOT(conf->hwtable);
 
+        if (!hwe)
+                return 1;
+
 	buff = set_value(strvec);
 	if (!buff)
 		return 1;
@@ -1021,6 +1027,9 @@ hw_pgpolicy_handler(vector strvec)
 	char * buff;
 	struct hwentry * hwe = VECTOR_LAST_SLOT(conf->hwtable);
 
+        if (!hwe)
+                return 1;
+
 	buff = set_value(strvec);
 
 	if (!buff)
