---
 libmultipath/hwtable.c  |   30 ++++++++++++++++++++++++++++++
 multipath.conf.defaults |   26 ++++++++++++++++++++++++++
 2 files changed, 56 insertions(+)

Index: multipath-tools-130222/libmultipath/hwtable.c
===================================================================
--- multipath-tools-130222.orig/libmultipath/hwtable.c
+++ multipath-tools-130222/libmultipath/hwtable.c
@@ -772,6 +772,36 @@ static struct hwentry default_hw[] = {
 		.prio_name     = PRIO_RDAC,
 		.prio_args     = NULL,
 	},
+	{
+		/* DELL MD36xxi */
+		.vendor        = "DELL",
+		.product       = "MD36xxi",
+		.bl_product    = "Universal Xport",
+		.features      = "2 pg_init_retries 50",
+		.hwhandler     = "1 rdac",
+		.pgpolicy      = GROUP_BY_PRIO,
+		.pgfailback    = -FAILBACK_IMMEDIATE,
+		.rr_weight     = RR_WEIGHT_NONE,
+		.no_path_retry = 15,
+		.checker_name  = RDAC,
+		.prio_name     = PRIO_RDAC,
+		.prio_args     = NULL,
+	},
+	{
+		/* DELL MD36xxf */
+		.vendor        = "DELL",
+		.product       = "MD36xxf",
+		.bl_product    = "Universal Xport",
+		.features      = "2 pg_init_retries 50",
+		.hwhandler     = "1 rdac",
+		.pgpolicy      = GROUP_BY_PRIO,
+		.pgfailback    = -FAILBACK_IMMEDIATE,
+		.rr_weight     = RR_WEIGHT_NONE,
+		.no_path_retry = 15,
+		.checker_name  = RDAC,
+		.prio_name     = PRIO_RDAC,
+		.prio_args     = NULL,
+	},
 	/*
 	 * NETAPP controller family
 	 *
Index: multipath-tools-130222/multipath.conf.defaults
===================================================================
--- multipath-tools-130222.orig/multipath.conf.defaults
+++ multipath-tools-130222/multipath.conf.defaults
@@ -655,6 +655,32 @@
 #		no_path_retry 15
 #	}
 #	device {
+#		vendor "DELL"
+#		product "MD36xxi"
+#		product_blacklist "Universal Xport"
+#		path_grouping_policy "group_by_prio"
+#		path_checker "rdac"
+#		features "2 pg_init_retries 50"
+#		hardware_handler "1 rdac"
+#		prio "rdac"
+#		failback "immediate"
+#		rr_weight "uniform"
+#		no_path_retry 15
+#	}
+#	device {
+#		vendor "DELL"
+#		product "MD36xxf"
+#		product_blacklist "Universal Xport"
+#		path_grouping_policy "group_by_prio"
+#		path_checker "rdac"
+#		features "2 pg_init_retries 50"
+#		hardware_handler "1 rdac"
+#		prio "rdac"
+#		failback "immediate"
+#		rr_weight "uniform"
+#		no_path_retry 15
+#	}
+#	device {
 #		vendor "NETAPP"
 #		product "LUN.*"
 #		path_grouping_policy "group_by_prio"
