---
 libmultipath/defaults.h       |    4 ++--
 multipathd/multipathd.service |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

Index: multipath-tools-130222/libmultipath/defaults.h
===================================================================
--- multipath-tools-130222.orig/libmultipath/defaults.h
+++ multipath-tools-130222/libmultipath/defaults.h
@@ -26,8 +26,8 @@
 #define MAX_CHECKINT(a)		(a << 2)
 
 #define MAX_DEV_LOSS_TMO	0x7FFFFFFF
-#define DEFAULT_PIDFILE		"/var/run/multipathd/multipathd.pid"
-#define DEFAULT_TIMESTAMP_FILE	"/var/run/multipathd/timestamp"
+#define DEFAULT_PIDFILE		"/run/multipathd/multipathd.pid"
+#define DEFAULT_TIMESTAMP_FILE	"/run/multipathd/timestamp"
 #define DEFAULT_SOCKET		"/org/kernel/linux/storage/multipathd"
 #define DEFAULT_CONFIGFILE	"/etc/multipath.conf"
 #define DEFAULT_BINDINGS_FILE	"/etc/multipath/bindings"
Index: multipath-tools-130222/multipathd/multipathd.service
===================================================================
--- multipath-tools-130222.orig/multipathd/multipathd.service
+++ multipath-tools-130222/multipathd/multipathd.service
@@ -9,7 +9,7 @@ Conflicts=shutdown.target
 
 [Service]
 Type=forking
-PIDFile=/var/run/multipathd/multipathd.pid
+PIDFile=/run/multipathd/multipathd.pid
 ExecStartPre=/sbin/modprobe dm-multipath
 ExecStartPre=-/sbin/multipath -A
 ExecStart=/sbin/multipathd
