---
 libmultipath/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: multipath-bz1296979/libmultipath/Makefile
===================================================================
--- multipath-bz1296979.orig/libmultipath/Makefile
+++ multipath-bz1296979/libmultipath/Makefile
@@ -30,7 +30,7 @@ ifneq ($(strip $(LIBDM_API_COOKIE)),0)
 	CFLAGS += -DLIBDM_API_COOKIE
 endif
 
-LIBUDEV_API_RECVBUF = $(shell grep -Ecs '^[a-z]*[[:space:]]+udev_monitor_set_resolve_buffer_size' /usr/include/libudev.h)
+LIBUDEV_API_RECVBUF = $(shell grep -Ecs '^[a-z]*[[:space:]]+udev_monitor_set_receive_buffer_size' /usr/include/libudev.h)
 
 ifneq ($(strip $(LIBUDEV_API_RECVBUF)),0)
 	CFLAGS += -DLIBUDEV_API_RECVBUF
