---
 multipathd/main.c |    2 ++
 1 file changed, 2 insertions(+)

Index: multipath-7.2.z/multipathd/main.c
===================================================================
--- multipath-7.2.z.orig/multipathd/main.c
+++ multipath-7.2.z/multipathd/main.c
@@ -1257,6 +1257,8 @@ check_path (struct vectors * vecs, struc
 	newstate = path_offline(pp);
 	if (newstate == PATH_UP)
 		newstate = get_state(pp, 1);
+	else
+		checker_clear_message(&pp->checker);
 
 	if (newstate == PATH_WILD || newstate == PATH_UNCHECKED) {
 		condlog(2, "%s: unusable path", pp->dev);
