---
 kpartx/dos.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: multipath-tools-130222/kpartx/dos.c
===================================================================
--- multipath-tools-130222.orig/kpartx/dos.c
+++ multipath-tools-130222/kpartx/dos.c
@@ -78,7 +78,7 @@ read_dos_pt(int fd, struct slice all, st
 	unsigned long offset = all.start;
 	int i, n=4;
 	unsigned char *bp;
-	int sector_size_mul = get_sector_size(fd)/512;
+	uint64_t  sector_size_mul = get_sector_size(fd)/512;
 
 	bp = (unsigned char *)getblock(fd, offset);
 	if (bp == NULL)
