---
 multipathd/main.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: multipath-tools-130222/multipathd/main.c
===================================================================
--- multipath-tools-130222.orig/multipathd/main.c
+++ multipath-tools-130222/multipathd/main.c
@@ -1248,7 +1248,8 @@ int update_prio(struct path *pp, int ref
 		return changed;
 	}
 	oldpriority = pp->priority;
-	pathinfo(pp, conf->hwtable, DI_PRIO);
+	if (pp->state != PATH_DOWN)
+		pathinfo(pp, conf->hwtable, DI_PRIO);
 
 	if (pp->priority == oldpriority)
 		return 0;
