---
 libmultipath/hwtable.c |   15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

Index: multipath-tools-130222/libmultipath/hwtable.c
===================================================================
--- multipath-tools-130222.orig/libmultipath/hwtable.c
+++ multipath-tools-130222/libmultipath/hwtable.c
@@ -65,14 +65,17 @@ static struct hwentry default_hw[] = {
 		.vendor        = "3PARdata",
 		.product       = "VV",
 		.features      = DEFAULT_FEATURES,
-		.hwhandler     = DEFAULT_HWHANDLER,
-		.pgpolicy      = MULTIBUS,
-		.pgfailback    = FAILBACK_UNDEF,
+		.hwhandler     = "1 alua",
+		.pgpolicy      = GROUP_BY_PRIO,
+		.pgfailback    = -FAILBACK_IMMEDIATE,
 		.rr_weight     = RR_WEIGHT_NONE,
-		.no_path_retry = 12,
-		.checker_name  = DEFAULT_CHECKER,
-		.prio_name     = DEFAULT_PRIO,
+		.no_path_retry = 18,
+		.checker_name  = TUR,
+		.prio_name     = PRIO_ALUA,
 		.prio_args     = NULL,
+		.selector      = "service-time 0",
+		.fast_io_fail  = 10,
+		.dev_loss      = MAX_DEV_LOSS_TMO,
 	},
 	{
 		.vendor        = "DEC",
