---
 multipathd/main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: multipath-tools-130222/multipathd/main.c
===================================================================
--- multipath-tools-130222.orig/multipathd/main.c
+++ multipath-tools-130222/multipathd/main.c
@@ -471,7 +471,7 @@ uev_add_path (struct uevent *uev, struct
 
 	pp = find_path_by_dev(vecs->pathvec, uev->kernel);
 	if (pp) {
-		condlog(0, "%s: spurious uevent, path already in pathvec",
+		condlog(2, "%s: spurious uevent, path already in pathvec",
 			uev->kernel);
 		if (pp->mpp)
 			return 0;
