---
 libmultipath/config.c |    3 ++-
 multipath/mpathconf.8 |    8 ++++++--
 2 files changed, 8 insertions(+), 3 deletions(-)

Index: multipath-tools-130222/libmultipath/config.c
===================================================================
--- multipath-tools-130222.orig/libmultipath/config.c
+++ multipath-tools-130222/libmultipath/config.c
@@ -739,7 +739,8 @@ load_config (char * file, struct udev *u
 		condlog(0, "/etc/multipath.conf does not exist, blacklisting all devices.");
 		condlog(0, "A default multipath.conf file is located at");
 		condlog(0, "/usr/share/doc/device-mapper-multipath-%d.%d.%d/multipath.conf", MULTIPATH_VERSION(VERSION_CODE));
-		condlog(0, "You can run /sbin/mpathconf to create or modify /etc/multipath.conf");
+		condlog(0, "You can run /sbin/mpathconf --enable to create");
+		condlog(0, "/etc/multipath.conf. See man mpathconf(8) for more details");
 		if (conf->blist_devnode == NULL) {
 			conf->blist_devnode = vector_alloc();
 			if (!conf->blist_devnode) {
Index: multipath-tools-130222/multipath/mpathconf.8
===================================================================
--- multipath-tools-130222.orig/multipath/mpathconf.8
+++ multipath-tools-130222/multipath/mpathconf.8
@@ -19,7 +19,9 @@ daemon, and configure the
 .B multipathd
 service to start automatically or not. If
 .B mpathconf
-is called with no commands, it will display the current configuration.
+is called with no commands, it will display the current configuration, but
+will not create or modify 
+.B /etc/multipath.conf
 
 The default options for mpathconf are
 .B --with_module
@@ -50,7 +52,9 @@ command.
 .B --enable
 Removes any line that blacklists all device nodes from the
 .B /etc/multipath.conf
-blacklist section.
+blacklist section. Also, creates
+.B /etc/multipath.conf
+if it doesn't exist.
 .TP
 .B --disable
 Adds a line that blacklists all device nodes to the
