---
 libmultipath/checkers/readsector0.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: multipath-tools-130222/libmultipath/checkers/readsector0.c
===================================================================
--- multipath-tools-130222.orig/libmultipath/checkers/readsector0.c
+++ multipath-tools-130222/libmultipath/checkers/readsector0.c
@@ -29,7 +29,7 @@ int libcheck_check (struct checker * c)
 	unsigned char sbuf[SENSE_BUFF_LEN];
 	int ret;
 
-	ret = sg_read(c->fd, &buf[0], 4069, &sbuf[0],
+	ret = sg_read(c->fd, &buf[0], 4096, &sbuf[0],
 		      SENSE_BUFF_LEN, c->timeout);
 
 	switch (ret)
