---
 libmultipath/discovery.c |    5 ++++-
 libmultipath/discovery.h |    2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

Index: multipath-tools-130222/libmultipath/discovery.c
===================================================================
--- multipath-tools-130222.orig/libmultipath/discovery.c
+++ multipath-tools-130222/libmultipath/discovery.c
@@ -749,7 +749,10 @@ do_inq(int sg_fd, int cmddt, int evpd, u
 	io_hdr.dxferp = resp;
 	io_hdr.cmdp = inqCmdBlk;
 	io_hdr.sbp = sense_b;
-	io_hdr.timeout = DEF_TIMEOUT;
+	if (conf->checker_timeout)
+		io_hdr.timeout = conf->checker_timeout * 1000;
+	else
+		io_hdr.timeout = DEF_TIMEOUT;
 
 	if (ioctl(sg_fd, SG_IO, &io_hdr) < 0)
 		return -1;
Index: multipath-tools-130222/libmultipath/discovery.h
===================================================================
--- multipath-tools-130222.orig/libmultipath/discovery.h
+++ multipath-tools-130222/libmultipath/discovery.h
@@ -14,7 +14,7 @@
 #endif
 
 #ifndef DEF_TIMEOUT
-#define DEF_TIMEOUT	300000
+#define DEF_TIMEOUT	60000
 #endif
 
 /*
