From 21136f36a38ce761a76797a2434c4fbbcbc89830 Mon Sep 17 00:00:00 2001
From: Hannes Reinecke <hare@suse.de>
Date: Mon, 2 Mar 2015 12:18:23 +0100
Subject: [PATCH] Add HP MSA 2040 to the hardware table

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 libmultipath/hwtable.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/libmultipath/hwtable.c b/libmultipath/hwtable.c
index a4ae053..28ee595 100644
--- a/libmultipath/hwtable.c
+++ b/libmultipath/hwtable.c
@@ -175,6 +175,21 @@ static struct hwentry default_hw[] = {
 		.prio_name     = PRIO_ALUA,
 		.prio_args     = NULL,
 	},
+	{
+		/* HP MSA 1040/2040 product family */
+		.vendor        = "HP",
+		.product       = "MSA (1|2)040 SA(N|S)",
+		.features      = DEFAULT_FEATURES,
+		.hwhandler     = DEFAULT_HWHANDLER,
+		.pgpolicy      = GROUP_BY_PRIO,
+		.pgfailback    = -FAILBACK_IMMEDIATE,
+		.rr_weight     = RR_WEIGHT_NONE,
+		.no_path_retry = 18,
+		.minio         = 100,
+		.checker_name  = TUR,
+		.prio_name     = PRIO_ALUA,
+		.prio_args     = NULL,
+	},
 
 	{
 		/* HP SVSP */
-- 
2.7.4

